sphinx/application.py

```diff
@@ -225,6 +226,7 @@ def __init__(self, srcdir, confdir, outdir, doctreedir, buildername,
         self.builder = self.create_builder(buildername)
         # check all configuration values for permissible types
         self.config.check_types()
+        self.emit('config-inited', self.config)
```

Comment:
This event must be emitted just after `config.init_values()`.

With current implementation, builders can touch config values before `config-inited` handlers invoked.